[ADD] accounting: l10n_br avalara implementation (#5199)
odoo/documentation: content/applications/finance/fiscal_localizations/brazil.rst

   * - :guilabel:`Brazil - Accounting Reports`
     - `l10n_br_reports`
     - Adds a simple tax report that helps check the tax amount per tax group in a given period of
       time. Also adds the P&L and BS adapted for the Brazilian market.
   * - :guilabel:`Avatax Brazil`
     - `l10n_br_avatax`
     - Add Brazilian tax calculation via Avatax and all necessary fields needed to configure Odoo in
       order to properly use Avatax and send the needed fiscal information to retrieve the correct
       taxes.
   * - :guilabel:`Avatax for SOs in Brazil`
     - `l10n_br_avatax_sale`
     - Same as the `l10n_br_avatax` module with the extension to the sales order module.

.. _brazil/company:

Configure your company
----------------------

To configure your company information, go to the :menuselection:`Contacts` app and search the name
given to your company.

#. Select the :guilabel:`Company` option at the top of the page. Then, configure the following
   fields:

   - :guilabel:`Name`
   - :guilabel:`Address` (add :guilabel:`City`, :guilabel:`State`, :guilabel:`Zip Code`,
     :guilabel:`Country`)
   - Tax ID (:guilabel:`CNPJ`)

>    - :guilabel:`IE`
>    - :guilabel:`IM`
>    - :guilabel:`SUFRAMA code` (if applicable)

Reviewer: ```suggestion
   - :guilabel:`IE` (State Registration)
   - :guilabel:`IM` (Municipal Registration)
   - :guilabel:`SUFRAMA code` (Superintendence of the Manaus Free Trade Zone - if applicable)
```

@samueljlieber maybe something like this. Please feel free to improve it if needed

       time. Also adds the P&L and BS adapted for the Brazilian market.
   * - :guilabel:`Avatax Brazil`
     - `l10n_br_avatax`
     - Add Brazilian tax calculation via Avatax and all necessary fields needed to configure Odoo in
       order to properly use Avatax and send the needed fiscal information to retrieve the correct
       taxes.
   * - :guilabel:`Avatax for SOs in Brazil`
     - `l10n_br_avatax_sale`
     - Same as the `l10n_br_avatax` module with the extension to the sales order module.

.. _brazil/company:

Configure your company
----------------------

To configure your company information, go to the :menuselection:`Contacts` app and search the name
given to your company. Alternatively, activate :ref:`developer mode <developer-mode>`, and go to
:menuselection:`Company --> Contact`. Then edit the contact.

#. Select the :guilabel:`Company` option at the top of the page. Then, configure the following
   fields:

   - :guilabel:`Name`
   - :guilabel:`Address` (add :guilabel:`City`, :guilabel:`State`, :guilabel:`Zip Code`,
     :guilabel:`Country`)
   - Tax ID (:guilabel:`CNPJ`)
   - :guilabel:`IE`
   - :guilabel:`IM`

>    - :guilabel:`SUFRAMA code` (if applicable)

Reviewer: They are common knowledge in BR.
But we can expand the abbreviations:
1. IE (State Registration)
2. IM (Municipal Registration)
3. SUFRAMA (Superintendence of the Manaus Free Trade Zone)

   - :guilabel:`Phone`
   - :guilabel:`Email`

   .. image:: brazil/company-configuration.png
      :alt: Company configuration.

#. Configure the :guilabel:`Fiscal Information` within the :guilabel:`Sales and Purchase` tab:

   - Add the :guilabel:`Fiscal Position` for :ref:`Avatax Brazil <brazil/fiscal-positions>`.

>    - :guilabel:`Tax Regime`
>    - :guilabel:`ICMS Taxpayer Type`
>    - :guilabel:`Main Activity Sector`

Reviewer: ```suggestion
   - :guilabel:`Tax Regime` (Federal Tax Regime)
   - :guilabel:`ICMS Taxpayer Type` (indicates ICMS regime, Exempt status, or Non-Taxpayer.)
   - :guilabel:`Main Activity Sector`
```

@samueljlieber here are some suggestions, please feel free to improve it if needed.
For the last one "Main Activity Sector" the label is already the description of what it is, so no further clarifications are needed there)

   .. image:: brazil/contact-fiscal-configuration.png
     :alt: Company fiscal configuration.

#. Finally, upload a company logo and save the contact.

.. note::

>    If you are a simplified regime you need to configure the ICMS rate in

Reviewer: Yes. agree with this change.
Let's use "_under_" instead of "_in_".
@samueljlieber can you apply this change?

~~~~~~~~~~~~~~~~~~~~~~~~

To activate AvaTax in Odoo, you need to create an account. To do so, go to
:menuselection:`Accounting --> Configuration --> Settings --> Taxes`, and, in the :guilabel:`AvaTax
Brazil` section, add the email address you want to use to log in to the AvaTax portal, and the click
on :guilabel:`Create account`. This email is used as the administrator email address in AvaTax.

After you create the account from Odoo, you need to go to the Avalara Portal to set up your
password.

.. warning::
   Only create an account with the AvaTax portal email you want to use in production when you are in
   your production database (real transactions). If you want to test, you can use a fake email to
   create the account with credentials for you to test on a testing or sandbox database.

.. tip::
   If you do not know how to log in to the AvaTax portal for the first time, `click here
   <https://drive.google.com/file/d/1jPtKzilfLrfdRXyPGacQNr6i2zvQvoOd/view?usp=sharing>`_. You can
   start using AvaTax in Odoo without creating a password and accessing the Avalara Portal; for
   Odoo, the only requirement to start using the Avalara Tax Computation Engine is to create an

>    account from the settings page.

Reviewer 2: It doesn't matter if something has to be configured outside of Odoo; we can - and sometimes should - still document processes outside of Odoo. Unless the external 'provider' documents it better (assuming here it would be better maintained) and/or it isn't necessary to configure the feature (i.e., extra info).
If the information makes too much noise, we don't need a link to an external document.

Here the information comes from us, meaning we could put it in the documentation instead. Furthermore, the information is relatively short at its core (the screenshots aren't needed):
> 1. Login on the [Avalara portal](https://portal.avalarabrasil.com.br/Login)
> 2. Click on “Meu primeiro acesso”
> 3. Add the email address you used in Odoo to create the Avalara/Avatax Account, and then click "Solicitar Senha"
> 4. You will receive an email with a Token and a link to create your password. Click on this link and copy-paste the token and allocate your desired password.

If you feel this information is superfluous, feel free not to include it, but I would avoid linking to an inconvenient and no so good-looking PDF.

.. image:: brazil/avatax-account-configuration.png
   :alt: Avatax account configuration.

.. note::

>    You can transfer API credentials. Use this only when you already created an account in another
>    Odoo instance and wish to reuse it.

Reviewer 2: ```suggestion
   You can transfer API credentials. Use this only when you have already created an account in another
   Odoo instance and wish to reuse it.
```

Configure master data
---------------------

Chart of accounts
~~~~~~~~~~~~~~~~~

The :doc:`chart of accounts </applications/finance/accounting/get_started/chart_of_accounts>` is

> installed by default as part of the set of data included in the localization module. The accounts

Reviewer 2: ```suggestion
installed by default as part of the data set included in the localization module. The accounts
```

are mapped automatically in their corresponding taxes, and the default account payable and account
receivable fields.

.. note::
   The chart of accounts for Brazil is based on the SPED CoA, which gives a baseline of the accounts
   needed in Brazil.

   You can add or delete accounts according to the company's needs.

Taxes
~~~~~

Taxes are automatically created when installing the Brazilian localization. Taxes are already
configured, and some of them are used by Avalara when computing taxes on the sales order or invoice.

> Taxes can be edited or more taxes can be added. For example, some taxes used for services need to be

Reviewer 2: ```suggestion
Taxes can be edited, or more taxes can be added. For example, some taxes used for services need to be
```

manually added and configured, as the rate may differ depending on the city where you are offering
the service.

.. important::
   Taxes attached to services are not computed by AvaTax. Only goods taxes are computed.

When configuring a tax used for a service that is included in the final price (when the tax is not
added or subtracted on top of the product price), set the :guilabel:`Tax Computation` to
:guilabel:`Percentage of Price Tax Included`, and, on the :guilabel:`Advanced Options` tab, check
the :guilabel:`Included in Price` option.

> For more information on how to configure taxes to better fit your needs, please go to the

Reviewer 2: ```suggestion
For more information on configuring taxes to fit your needs better, please go to the
```

:doc:`taxes functional documentation </applications/finance/accounting/taxes>`.

.. image:: brazil/tax-configuration.png
   :alt: Tax configuration.

.. warning::
   Do not delete taxes, as they are used for the AvaTax tax computation. If deleted, Odoo creates
   them again when used in an |SO| or invoice and computing taxes with AvaTax, but the account used
   to register the tax needs to be re-configured in the tax's :guilabel:`Definition` tab, under the
   :guilabel:`Distribution for invoices` and :guilabel:`Distribution for refunds` sections.

.. _brazil/products:

Products
~~~~~~~~

To use the AvaTax integration on sale orders and invoices, first specify the following information
on the product:

- :guilabel:`CEST Code`
- :guilabel:`Mercosul NCM Code`
- :guilabel:`Source of Origin`
- :guilabel:`SPED Fiscal Product Type`

> - :guilabel:`Purpose of Use`

Reviewer 2: More info on filling out these fields could be useful

- :guilabel:`Sales Price`

.. image:: brazil/product-configuration.png
   :alt: Product configuration.

.. note::
   Odoo automatically creates three products to be used for transportation costs associated with

>    sales. These are named `Freight`, `Insurance`, and `Other Costs`, they are already configured, if
>    more need to be created, duplicate and use the same configuration (configuration needed:

Reviewer 2: ```suggestion
   sales. These are named `Freight`, `Insurance`, and `Other Costs`. They are already configured, and if
   more need to be created, duplicate and use the same configuration (configuration needed:
```

   :guilabel:`Product Type` `Service`, :guilabel:`Transportation Cost Type` `Insurance`,
   `Freight`, or `Other Costs`)

.. _brazil/contacts:

Contacts
~~~~~~~~

Before using the integration, specify the following information on the contact:

#. General information about the contact:

   - Select the :guilabel:`Company` option for a contact with a tax ID (CNPJ), or check

>      :guilabel:`Individual` for a contact with a CPF

Reviewer 2: ```suggestion
     :guilabel:`Individual` for a contact with a CPF.
```

   - :guilabel:`Name`

>    - :guilabel:`Address`: :guilabel:`Zip Code` is a required field to properly compute taxes.

Reviewer 2: ```suggestion
   - :guilabel:`Address`: :guilabel:`Zip Code` is a required field to compute taxes properly.
```

>    - :guilabel:`Tax ID` or :guilabel:`CPF`: depending on the case complete with the number

Reviewer 2: unclear

Reviewer: ```suggestion
   - :guilabel:`Tax ID` or :guilabel:`CPF`: Use CPF for individuals and Tax ID (CNPJ) for companies
```
Also, in BR users know what these abbreviations refer to, but true that this is unclear.
@samueljlieber can you help me with this change?
Thanks!

   - :guilabel:`IE`: state tax identification number
   - :guilabel:`IM`: municipa tax identification number
   - :guilabel:`SUFRAMA code`: SUFRAMA registration number
   - :guilabel:`Phone`
   - :guilabel:`Email`

   .. image:: brazil/contact-configuration.png
     :alt: Contact configuration.

   .. note::
      The :guilabel:`CPF`, :guilabel:`IE`, :guilabel:`IM`, and :guilabel:`SUFRAMA code` fields are
      are hidden until the :guilabel:`Country` is set to `Brazil`.

#. Fiscal information about the contact under the :guilabel:`Sales & Purchase` tab:

>    - :guilabel:`Fiscal Position`: add the AvaTax fiscal position to automatically compute taxes on
>      sale orders and invoices

Reviewer 2: ```suggestion
   - :guilabel:`Fiscal Position`: add the AvaTax fiscal position to compute taxes on
     sale orders and invoices automatically.
```

   - :guilabel:`Tax Regime`: federal tax regime
   - :guilabel:`ICMS Taxpayer Type`: taxpayer type determines if the contact is within the ICMS

>      regime, if it is exempt, or if it is a non-taxpayer

Reviewer 2: ```suggestion
     regime, if it is exempt, or if it is a non-taxpayer.
```

   - :guilabel:`Main Activity Sector`: list of main activity sectors of the contact

   .. image:: brazil/contact-fiscal-configuration.png
     :alt: Contact fiscal configuration.

.. _brazil/fiscal-positions:

Fiscal positions
~~~~~~~~~~~~~~~~

To compute taxes on sale orders and invoices, it is necessary to have a :guilabel:`Fiscal Position`

> with the :guilabel:`Detect Automatically` and the :guilabel:`Use AvaTax API` selected.

Reviewer 2: ```suggestion
with the :guilabel:`Detect Automatically` and the :guilabel:`Use AvaTax API` options enabled.
```

The :guilabel:`Fiscal Position` can be configured on the contact or selected when creating a sales
order or an invoice.

.. image:: brazil/fiscal-position-configuration.png
   :alt: Fiscal position configuration

Workflows
=========

This section provides an overview of the actions that trigger API calls for tax computation.

.. warning::
   Please note that each API call incurs a cost. Be mindful of the actions that trigger these calls
   to manage costs effectively.

Tax calculations on quotation / sales orders
--------------------------------------------

> Trigger an API call to calculate taxes on a quotation or sales order automatically with AvaTax any

Reviewer 2: ```suggestion
Trigger an API call to calculate taxes on a quotation or sales order automatically with AvaTax in any
```

of the following ways:

- **Quotation confirmation**
    Confirm a quotation into a sales order.
- **Manual trigger**
    Click on :guilabel:`Compute Taxes Using Avatax`.
- **Preview**
    Click on the :guilabel:`Preview` button.
- **Email a quotation / sales order**
    Send a quotation or sales order to a customer via email.
- **Online quotation access**
    When a customer accesses the quotation online (via the portal view), the API call is triggered.

Tax calculations on invoices
----------------------------

Trigger an API call to calculate taxes on a customer invoice automatically with AvaTax any of the
following ways:

- **Manual trigger**
    Click on :guilabel:`Compute Taxes Using AvaTax`.
- **Preview**
    Click on the :guilabel:`Preview` button.
- **Online invoice access**
    When a customer accesses the invoice online (via the portal view), the API call is triggered.

.. note::
   The :guilabel:`Fiscal Position` must be set to `Automatic Tax Mapping (Avalara Brazil)` for any

>    of these actions to automatically compute taxes.

Reviewer 2: ```suggestion
   of these actions to compute taxes automatically.
```

For the services taxes, you can create and configure them from Odoo directly without the need of
computing them with AvaTax.

The localization also includes a chart of account and taxes created as a template that can be
modified if needed.

Configuration
=============

Modules installation
--------------------

:ref:`Install <general/install>` the following modules to get all the features of the Brazilian
localization:

.. list-table::
   :header-rows: 1
   :widths: 25 25 50

   * - Name
     - Technical name
     - Description
   * - :guilabel:`Brazilian - Accounting`
     - `l10n_br`
     - Default :ref:`fiscal localization package <fiscal_localizations/packages>` adds accounting
       characteristics for the Brazilian localization, which represent the minimum configuration
       required for a company to operate in Brazil. The module's installation automatically loads:
       chart of accounts, taxes, and include required fields to properly configure the contact.
   * - :guilabel:`Brazil - Accounting Reports`
     - `l10n_br_reports`
     - Includes all the technical and functional requirements to generate and validate
       :doc:`electronic documents

>        </applications/finance/accounting/customer_invoices/electronic_invoicing>`, based on the
>        technical documentation published by the SRI. The authorized documents are: invoices, credit
>        notes, debit notes, withholdings and purchase liquidations.

Reviewer: ```suggestion
       </applications/finance/accounting/customer_invoices/electronic_invoicing>`, adds a simple Tax Report that helps check the tax amount per tax group in a given period of time. Also adds the P&L and BS adapted for the Brazilian market.
```

Can you help with this correction @samueljlieber as I made a mistake and kept with the wrong info here.
Thanks!

>      - Includes all the technical and functional requirements to generate and validate
>        :doc:`electronic documents
>        </applications/finance/accounting/customer_invoices/electronic_invoicing>`, based on the
>        technical documentation published by the SRI. The authorized documents are: invoices, credit
>        notes, debit notes, withholdings and purchase liquidations.

Reviewer: ```suggestion
     - Adds a simple Tax Report that helps check the tax amount per tax group in a given period of time. Also adds the P&L and BS adapted for the Brazilian market.
```

Can you help with this correction @samueljlieber as I made a mistake and kept with the wrong info here.
Thanks!

To configure your company information, go to the :menuselection:`Contacts` app and search the name
given to your company or activate :ref:`developer mode <developer-mode>` and go to
:menuselection:`Company --> Contact` and then edit the contact.

1. Select the :guilabel:`Company` option at the top of the page. Then, configure the following
   fields:

   - :guilabel:`Name`
   - :guilabel:`Address` (add :guilabel:`City`, :guilabel:`State`, :guilabel:`Zip Code`,
     :guilabel:`Country`)
   - Tax ID (:guilabel:`CNPJ`)
   - :guilabel:`IE`
   - :guilabel:`IM`
   - :guilabel:`SUFRAMA code` (if applicable)
   - :guilabel:`Phone`
   - :guilabel:`Email`

.. image:: brazil/company-configuration.png
  :align: center
  :alt: Company configuration.

2. Configure the :guilabel:`Fiscal Information` within the :guilabel:`Sales and Purchase` tab:

   - Add the :guilabel:`Fiscal Position` for :ref:`Avatax Brazil <brazil/fiscal-positions>`.
   - :guilabel:`Tax Regime`
   - :guilabel:`ICMS Taxpayer Type`
   - :guilabel:`Main Activity Sector`

.. image:: brazil/contact-fiscal-configuration.png
  :align: center
  :alt: Company fiscal configuration.

Finally, upload a company logo and save the contact.

> 

Reviewer: @samueljlieber can we also add an additional Note here below saying:
If you are a Simplified Regime you need to configure the ICMS rate in :menuselection:`Accounting
--> Configuration --> Settings --> Taxes --> Avatax Brazil` 

Thanks!

   Odoo instance and wish to reuse it.

Configure master data
---------------------

Chart of accounts
~~~~~~~~~~~~~~~~~

The :doc:`chart of accounts </applications/finance/accounting/get_started/chart_of_accounts>` is
installed by default as part of the set of data included in the localization module, the accounts
are mapped automatically in taxes, default account payable, and default account receivable.

The chart of accounts for Brazil is based on the SPED CoA, this gives a baseline of the accounts
needed in Brazil.

You can add or delete accounts according to the company's needs.

Taxes
~~~~~

Taxes will be automatically created when installing the Brazilian localization. Taxes will be
already configured, and some of them will be used by Avalara when computing taxes on the sale order
or invoice.

Taxes can be edited or more taxes can be added, for example, some taxes used for services need to be
manually added and configure, as depending on the city that you are offering the service, the rate
may differ, and **taxes attached to services will not be computed by Avatax**, only good taxes are
computed.

> When adding a tax used for a service that is included in the final price - when the tax is not added
> or subtracted on top of the product price - then you need to configure the :guilabel:`Tax
> Computation` to :guilabel:`Percentage of Price Tax Included`, and on the :guilabel:`Advanced
> Options` tab check the :guilabel:`Included in Price` option.

Reviewer: ```suggestion
When adding a tax used for a service that is included in the final price - when the tax is not added
or subtracted on top of the product price - then you need to configure the :guilabel:`Tax
Computation` to :guilabel:`Percentage of Price Tax Included`, and on the :guilabel:`Advanced
Options` tab check the :guilabel:`Included in Price` option.

For more information on how to configure taxes to better fit your needs, please go to the [taxes functional documentation](https://www.odoo.com/documentation/16.0/applications/finance/accounting/taxes.html#taxes)
```

Reviewer: @samueljlieber please help me add this also here.
Thanks!

> .. image:: brazil/tax-configuration.png

Reviewer: @samueljlieber also this image is wrong, I retook the screenshot.
Here is the one to be used here: https://drive.google.com/file/d/1w0JnmUjeY_iycauVkPyYUZ90bKKpehbK/view?usp=sharing

Sorry for that.
